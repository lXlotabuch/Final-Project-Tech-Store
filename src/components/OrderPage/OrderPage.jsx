import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { useHistory, Link } from 'react-router-dom';
import { selectIsLogin } from '../../store/auth/reducer';
import { RowColumn, WrapperButton } from '../CartPage/Flex';
import StyledButton from '../common/Buttons/StyledButton';
import { ContainerCart } from '../common/Container';
import Heading from '../common/Heading/Heading';
import OrderInfo from './OrderInfo/OrderInfo';

export const OrderPageComponent = ({isLogin}) => {
  useEffect(() => {
    window.scrollTo(0, 0)
  }, [])
  
  const history = useHistory()

  const onClickHome = () => {
    history.push('/')
  }
  const onClickDashboard = () => {
    history.push('/dashboard')
  }
  return (
    <ContainerCart>
      <RowColumn>
        <Heading>Order</Heading>
        <WrapperButton>
          <StyledButton
            onClick={onClickHome}
            size="lg"
            shape="round"
            color="borderBlue"
          >
            Back to Homepage
          </StyledButton>
        </WrapperButton>
        
        {isLogin
          ? (
            <WrapperButton>
              <Link to="/dashboard">
                <StyledButton
                  onClick={onClickDashboard}
                  size="lg"
                  shape="round"
                  color="borderGrey"
                >
                  Go to Dashboard
                </StyledButton>
              </Link>
            </WrapperButton>
          )
          : null}
      </RowColumn>
      <OrderInfo />
    </ContainerCart>
  )
}

const mapStateToProps = (state) => ({
  isLogin: selectIsLogin(state)
})

const OrderPage = connect(mapStateToProps, null)(OrderPageComponent)

OrderPageComponent.propTypes = {
  isLogin: PropTypes.bool.isRequired,
}

export default OrderPage