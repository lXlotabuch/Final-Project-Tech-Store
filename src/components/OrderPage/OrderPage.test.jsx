/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Router } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { OrderPageComponent } from './OrderPage';

jest.mock('./OrderInfo/OrderInfo', () => 'order-info')

describe('OrderPageComponent', () => {
  test('render', () => {
    const history = createMemoryHistory()
    const {asFragment} = render(
      <Router history={history}>
        <OrderPageComponent
          isLogin={false}
        />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('handle go to homePage', () => {
    const onClickHome = jest.fn()
    const history = createMemoryHistory()
    render(
      <Router history={history}>
        <OrderPageComponent
          isLogin={false}
          onClickHome={onClickHome}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /back to homepage/i}))
    expect(history.location.pathname).toBe('/')
  })

  test('handle go to dashboard', () => {
    const onClickDashboard = jest.fn()
    const history = createMemoryHistory()
    render(
      <Router history={history}>
        <OrderPageComponent
          isLogin
          onClickDashboard={onClickDashboard}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /go to dashboard/i}))
    expect(history.location.pathname).toBe('/dashboard')
  })
})