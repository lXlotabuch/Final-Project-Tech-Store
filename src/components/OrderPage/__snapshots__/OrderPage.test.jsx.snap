// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderPageComponent render 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw sc-pFZIQ jYBaYX bvRMiD"
  >
    <div
      class="sc-bdfBwQ sc-hKgILt dkvZGM ejUGax"
    >
      <h3
        class="sc-jrAGrp UbNtR"
      >
        Order
      </h3>
      <div
        class="sc-gKsewC ixmeZn"
      >
        <button
          class="ant-btn ant-btn-round sc-iBPRYJ iRkwl"
          color="borderBlue"
          type="button"
        >
          <span>
            Back to Homepage
          </span>
        </button>
      </div>
    </div>
    <order-info />
  </div>
</DocumentFragment>
`;
